File: week06/explore_phila_data/prepare_septa_gtfs.mjs
import * as csv from 'csv/sync';
import fs from 'fs/promises';
import path from 'path';
import { fileURLToPath } from 'url';

const __dirname = path.dirname(fileURLToPath(import.meta.url));
const RAW_DATA_DIR = path.join(__dirname, 'raw_data/');
const PREPARED_DATA_DIR = path.join(__dirname, 'prepared_data/');

const GTFS_FEEDS = ['septa_bus', 'septa_rail'];

for (const gtfsFeed of GTFS_FEEDS) {
  const gtfsFeedFolder = path.join(RAW_DATA_DIR, gtfsFeed + '/');
  for (const gtfsFileName of await fs.readdir(gtfsFeedFolder)) {

    // Read the data from the raw GTFS CSV file.
    const gtfsFilePath = gtfsFeedFolder + gtfsFileName;
    const content = await fs.readFile(gtfsFilePath, 'utf8');
    const rows = csv.parse(content, { "columns": true, "skip_empty_lines": true });

    // Write the data to a new prepared CSV file, creating the folder if it
    // doesn't exist.
    const outputFolder = path.join(PREPARED_DATA_DIR, gtfsFeed + '/');
    const outputPath = path.join(outputFolder, gtfsFileName.replace('.txt', '.jsonl'));
    await fs.mkdir(outputFolder, { recursive: true });

    await fs.writeFile(
      outputPath,
      rows.map(row => JSON.stringify(row)).join('\n'),
      'utf8'
    );
  }
}

File: week06/explore_phila_data/prepare_septa_gtfs.py
import csv
import json
import pathlib

RAW_DATA_DIR = pathlib.Path(__file__).parent / 'raw_data'
PREPARED_DATA_DIR = pathlib.Path(__file__).parent / 'prepared_data'

GTFS_FEEDS = ['septa_bus', 'septa_rail']

for gtfs_feed in GTFS_FEEDS:
    gtfs_feed_folder = RAW_DATA_DIR / gtfs_feed
    for gtfs_path in gtfs_feed_folder.iterdir():

        # Read the data from the raw GTFS CSV file.
        with gtfs_path.open('r', encoding='utf-8') as f:
            reader = csv.reader(f)
            rows = [row for row in reader]

        # Write the data to a new prepared CSV file, creating the
        # necessary directories if they don't exist.
        output_folder = PREPARED_DATA_DIR / gtfs_feed
        output_path = output_folder / f'{gtfs_path.stem}.csv'
        output_folder.mkdir(parents=True, exist_ok=True)

        with output_path.open('w', encoding='utf-8') as f:
            f.write('\n'.join(json.dumps(row) for row in rows))
